pkg/providers/vault/secret_manager.go

```diff
@@ -85,6 +85,11 @@ func (c HttpSecretStoreManager) getAllKeys() (map[string]interface{}, error) {
 	}
 
 	req.Header.Set(c.HttpConfig.Authentication.AuthType, c.HttpConfig.Authentication.AuthToken)
+
+	if c.HttpConfig.Namespace != ""{
+		req.Header.Set(NamespaceHeader, c.HttpConfig.Namespace)
```

Comment:
Did anyone notice that Namespaces are an  ENTERPRISE feature?

https://www.vaultproject.io/docs/enterprise/namespaces/index.html

I think we need use the namespace to build the path manually, rather than relying on the header.

Reply:
Good point, and great catch. I did not realize this was an enterprise only feature. I still like the idea of using the namespace as a namespace if it is provided. If the request happens to fail it should be logged by the `SecretClient` and clear as to what happened as it is reported by the server(Vault). The reason I favor this approach is it seems more deterministic to use the values as they are provided rather than concatenating multiple values to get around the lack of the namespace feature due to the configuration of Vault . I think we can document what the namespace is used for and that is a feature available for Vault enterprise.